### DISCUSSION TITLE
New navigation option "Show only first level of node"

### DISCUSSION THREAD

**Discussion by aaa1386** [freeplane/freeplane#1608] [Ideas]
Node A has branches B, C, D, E, F, G.
Suppose you want the nodes "B, C, D, E, F, G"
View them side by side so that they are all visible on the screen at once instead of scrolling.

See the ways in the video below:

Wouldn't it be better if there was an option to display only the first level of that node?
Especially since seeing the first level of a node is often necessary to have a better focus and is used a lot.

https://github.com/freeplane/freeplane/assets/38586259/399b8200-42c5-4819-9365-714edfaa22f6

https://github.com/freeplane/freeplane/assets/38586259/d9432371-6409-4702-8304-32d9c638ed48

https://github.com/freeplane/freeplane/assets/38586259/34a3648b-aada-4956-a5a2-9549e0066449

**Comment by aaa1386** [comment]
Especially the methods shown in the second video and the following video cause the mind map to fold in many nodes and sometimes the user does not want to change the open or closed state of the nodes.

https://github.com/freeplane/freeplane/assets/38586259/883f6468-db58-48ec-a46a-cd6db700a718

**Comment by dpolivaev** [comment]
Can someone check if the features aren't all available?

**Reply by aaa1386** [reply]
It is true that the software has many features.
But I gave this idea to improve the software.

My argument is:
1- Undoubtedly, the first level of each node is very important. In the mind of the user, the categories are remembered from more general to more detailed. On the software, the user first sees the nodes of the first level and navigates to the desired branch among them.
  When the first level is important, it should also be easy to access. In the videos that were sent, how many steps must be taken to be able to see the first level of a node. Sometimes it is confusing to decide which way to go.
I did not propose this idea to see the second, third and higher levels. Because the first level is the most important.

2- It is very important to see the nodes of the first level together. When the mental map is in wide mode and the distance between the nodes of the first level increases and some of them are not visible on the screen. As in the videos sent, nodes "B, C, D, E, F, G" are not all visible on the screen because they are far apart. So there should be an option to quickly view level one nodes side by side.

3- Many options of the software are very useful, for example, the" Navigate > (UN)Fold "option is very useful. Some options are less useful, such as the " Navigate > (UN)Fold Children" option.
  "Fold Children" option targets the second level. I haven't used this option even once while using the software. Other people's experience may be different. Although I have a strong guess that others don't use it much either.

Be sure that adding this idea to the software is one of the things that has a lot of use. Thanks .

**Comment by sifran-github** [comment]
I don't really understand your point.
you showed unfolding children (ctrl + space) in you 3rd video, which does exactly that. it will fold and show the immediate children.

you also state that "sometimes the user does not want to change the open or closed state of the nodes" but in your suggested "view of only children nodes", the user doesn't care about the state of nodes because he doesn't see them, only children... so... you seem to suggesting some view mode that will not fold nodes, although if you use it... you don't really care that its folding nodes? and this can be done right now easily using ctrl + space?

not sure that I follow the idea.
it needs to be more clear.

**Comment by aaa1386** [comment]
"(ctrl + space) or (UN)Fold Children" can only show the first level of the node in some cases. See the figure below. In this case, it is not possible to directly show the first level of the node with the "(UN)Fold Children" option.
Therefore, "Fold Children" can only do that directly in some cases, not always.

![image](https://github.com/freeplane/freeplane/assets/38586259/6eab034e-e81c-4d7d-9c40-92f838fb5bd0)

You might say that when the node is completely folded, you can use the option "unflood one level". This is also problematic. When this option is used, it will collapse all other layers. And such work means tampering with user settings. Watch the video below


https://github.com/freeplane/freeplane/assets/38586259/33e760e4-7f9d-4d41-b83d-0bf3ccb74aec


  In addition, it is confusing to use two different ways to represent the first level in different situations.
If this idea is implemented, you can easily create the first level in any situation. You don't need to think about how to do this with several intermediaries.

But the second thing: this idea does not change the user's settings. It means that the first level is folded, the second level is hidden, but its settings remain in the software memory. Exactly like the function of "Navigate>Fold>(UN)Fold" option. When the idea is implemented, the software should fold before level two, not for all levels.


The function of the proposed idea should be according to the "Navigate>Fold>(UN)Fold" function, so that it does not mess up the user's settings, see the following video:

https://github.com/freeplane/freeplane/assets/38586259/edb65646-0cca-48c8-839a-a342a69e95d5

**Reply by aaa1386** [reply]
Another example of the correct function in maintaining the open and closed settings of the sub-set nodes is the "Navigate > Fold > Fold Children" option. Because it only folds from children, not the next layers. Therefore, user settings are preserved.

**Comment by aaa1386** [comment]
It is good that the developer of the software fixes the function of "unfold one level". According to the form shown in the video. As a result, the user's settings change the least. Of course, if possible.

**Comment by aaa1386** [comment]
If someone can write a script, I would be grateful.
For the "Show only first level of node" option.

**Comment by aaa1386** [comment]
This idea can complement "Auto Extend"" VER 1.11.9_4". It is hard to explain.
One of the basics of the "auto extend" idea is the opening of the first level of a node. Sometimes the user manually unfolds a node completely. This makes the "Auto Extend" option not work on it. But if the user makes a new decision to see only the first level of that node, he can directly see the first level with the proposed idea.
This makes it possible to always keep "Auto Extend" active. The power of user navigation becomes great. It can also use other navigation modes and always access the first level by "Auto Extend" and the proposed idea easily. This means super strong and comfortable navigation power.

**Comment by aaa1386** [comment]
🌱🌱🌱
Summary of the idea:
1- Adding the "Show only first level of node" option
This option works in any state the node is in. For example, the situation where the knot is completely folded. It can work unlike the "Navigate > Fold > (un)Fold Children" option which doesn't work when the node is fully folded. In other situations, such as when several levels of the node are open, it can also show only the first level after its execution.

2- Fixing the bug of "unfold one level" option.
The drawback of this option is changing user settings in hidden levels. It was shown in one of the videos above. Only one option should fold all nodes, that is "Navigate > Fold > Fold all" option, not other options.

When the first level is shown. With the "unfold one level" option, the next levels can be displayed. Similar to "Navigate > Fold > Fold Children" function
3- Removing the "Navigate > Fold > Fold Children" option
Because it doesn't really have much use. If it is not removed, it will not affect the idea.

I thought a lot that this idea is much more practical and convenient than the current state of the software.

**Comment by aaa1386** [comment]
If this idea is implemented, I will propose another idea that is very interesting.
